src/uct/ib/efa/ib_efa_md.c

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <uct/ib/efa/ib_efa.h>
+#include <uct/ib/base/ib_md.h>
+#include <ucs/type/status.h>
+#include <ucs/debug/log.h>
+
+
+static uct_ib_md_ops_t uct_ib_efa_md_ops;
+
+static ucs_status_t uct_ib_efa_md_open(struct ibv_device *ibv_device,
+                                       const uct_ib_md_config_t *md_config,
+                                       uct_ib_md_t **p_md)
+{
+    struct ibv_context *ctx;
+    uct_ib_efadv_md_t *md;
+    struct efadv_device_attr attr;
+    ucs_status_t status;
+
+    ctx = ibv_open_device(ibv_device);
+    if (ctx == NULL) {
+        ucs_diag("ibv_open_device(%s) failed: %m",
+                 ibv_get_device_name(ibv_device));
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (efadv_query_device(ctx, &attr, sizeof(attr))) {
+        status = UCS_ERR_UNSUPPORTED;
+        goto err_free_context;
+    }
```

Comment:
```suggestion
    ret = efadv_query_device(ctx, &attr, sizeof(attr)); 
    if (ret != 0) {
        ucs_error("efadv_query_device failed: %d", ret);
        status = UCS_ERR_UNSUPPORTED;
        goto err_free_context;
    }
```

Reply:
fixed with `ucs_debug()` as this is called on non-EFA device too.